File: src/merchandise/router.ts
import { Router } from 'express';

import {
  handleGetMerchandises,
  handleGetMerchandise,
  handleCreateMerchandise,
  handleUpdateMerchandise,
  handleDeleteMerchandise,
} from './controller';
import authMiddleware from '../auth/middleware';

const merchandiseRouter = Router();

merchandiseRouter.get('/', handleGetMerchandises);
merchandiseRouter.get('/:id', handleGetMerchandise);
merchandiseRouter.post('/', authMiddleware, handleCreateMerchandise);
merchandiseRouter.put('/:id', authMiddleware, handleUpdateMerchandise);
merchandiseRouter.delete('/:id', authMiddleware, handleDeleteMerchandise);

export default merchandiseRouter;

File: merchandise/router.go
package merchandise

import (
	"github.com/AndreyGalchevski/strident-api/auth"
	"github.com/gorilla/mux"
)

func InitMerchandiseRouter(r *mux.Router) {
	r.HandleFunc("/merchandise", handleGetMerchandise).Methods("GET")

	r.HandleFunc("/merchandise/{id}", auth.VerifyAuthorization(handleGetMerchandiseByID)).Methods("GET")
	r.HandleFunc("/merchandise", auth.VerifyAuthorization(handlePostMerchandise)).Methods("POST")
	r.HandleFunc("/merchandise/{id}", auth.VerifyAuthorization(handlePatchMerchandise)).Methods("PATCH")
	r.HandleFunc("/merchandise/{id}", auth.VerifyAuthorization(handleDeleteMerchandise)).Methods("DELETE")
}
